### PR TITLE
Provide way to override title of home page

### PR DESCRIPTION
**Pull request by maelle**
Not sure I've learnt enough about `pkgdown` testing from my other PR, but here is my PR for #957 :-)

### CODE REVIEW

**Comment by jayhesselberth** [comment]
LGTM

File: tests/testthat/test-build-home-index.R [hunk 1]
```diff
@@ -25,3 +25,12 @@ test_that("package repo verification", {
   )
 })
 
+test_that("page title",{
```
Comment (hadley) [inline tests/testthat/test-build-home-index.R]:
```suggestion
test_that("page title can be overridden",{
```

File: tests/testthat/test-build-home-index.R [hunk 2]
```diff
@@ -25,3 +25,12 @@ test_that("package repo verification", {
...
+  pkg <- test_path("assets/pagetitle")
+  expect_equal(as.character(data_home(pkg))[[1]],
```
Comment (hadley) [inline tests/testthat/test-build-home-index.R]: Can you please use tidyverse style for long lines here? http://style.tidyverse.org/syntax.html#long-lines

File: tests/testthat/test-build-home-index.R [hunk 3]
```diff
@@ -25,3 +25,12 @@ test_that("package repo verification", {
...
+               "Such a cool package")
+
+  pkg <- test_path("assets/home-index-rmd")
```
Comment (hadley) [inline tests/testthat/test-build-home-index.R]: I think this should be in a separate test called something like "page title defaults to package title", and it should come before the test for the override.

File: tests/testthat/test-build-home-index.R [hunk 4]
```diff
@@ -25,3 +25,12 @@ test_that("package repo verification", {
   )
 })
 
+test_that("page title",{
+  pkg <- test_path("assets/pagetitle")
```
Comment (hadley) [inline tests/testthat/test-build-home-index.R]: It seems a bit weird to create a whole directory for this single test. Can you think of a simpler approach?

Comment (maelle) [reply tests/testthat/test-build-home-index.R]: Ok, but not sure the approach I use now is much better?

I thought of using mocking but then when looking at `as_pkgdown()` that I'd have mocked I saw it returned its argument if it were already a pkgdown object, so used that feature.

File: tests/testthat/test-build-home-index.R [hunk 5]
```diff
@@ -25,3 +25,22 @@ test_that("package repo verification", {
...
+
+test_that("page title can be overridden", {
+  pkg <- test_path("assets/home-index-rmd")
+  pkg <- as_pkgdown(pkg)
+  pkg$meta <- list(title = "Such a cool package")
+
+  expect_equal(
+    as.character(data_home(pkg))[[1]],
+    "Such a cool package"
+    )
```
Comment (hadley) [inline tests/testthat/test-build-home-index.R]:
```suggestion
  )
```

File: tests/testthat/test-build-home-index.R (hunk 4 above)
Comment (hadley) [reply tests/testthat/test-build-home-index.R]: Yes, this is much better. I wouldn't expect a large test when you're making such a small change to the code.

Comment (maelle) [reply tests/testthat/test-build-home-index.R]: Noted for next time! 😉

**Pull request merged**